### ISSUE TITLE
How to use a custom dataset with this model?

### ISSUE THREAD

**Issue by Vaibhav-ML** [yaoyao-liu/meta-transfer-learning#41]
Hi,
I have my own dataset. Folder Structure is as follows 

Data -
          Folder 1(contains images and csv,json file)
          Folder 2 (same as above)
         .
         .
         Folder N(contains images and csv,json file)

How do I use my own dataset with this model?

**Comment by yaoyao-liu** [comment]
Hi @Vaibhav-ML,

Sorry for the late reply.
If you need to run our code on your own dataset, you may revise the files in `meta-transfer-learning/tensorflow/data_generator` folder according to your dataset. If you have any further questions, feel free to add comments on this issue or email me.

Best,
Yaoyao

**Comment by Vaibhav-ML** [comment]
Hi Liu,
         I have another query. My dataset contains images of traffic. So
each image has multiple cars,trucks,cycles etc.

So, is it possible to detect multiple objects in a single image using your
model?

If yes, then exactly what code changes would you advise to do for this to
work?

Also, I have a json file which contains the bounding box info for the
objects. How can I use that?


Regards,
Vaibhav Sharma
TU Dresden

**Comment by yaoyao-liu** [comment]
Hi @Vaibhav-ML,

Our method cannot directly work on object detection tasks. As I am not working on object detection, I am not able to give you advice on that. Maybe you can follow the papers like [this one](https://openaccess.thecvf.com/content_ICCV_2019/papers/Kang_Few-Shot_Object_Detection_via_Feature_Reweighting_ICCV_2019_paper.pdf).

Best.
Yaoyao

**Comment by Vaibhav-ML** [comment]
Hi Liu,
           Thanks for your response. So you mean to say that this model can
only be used for image classification but not object detection right?

Regards,
Vaibhav Sharma
TU Dresden

**Comment by yaoyao-liu** [comment]
Exactly. Most of the few-shot learning papers evaluate their method on classification, some also provide results for reinforcement learning, e.g., MAML. However, I seldom see papers titled "few-shot learning" evaluate their method on the object detection task.  If they run experiments on object detection, they usually directly specify "few-shot object detection" in the title of the paper.
